packages/svelte/src/compiler/phases/3-transform/client/visitors/RegularElement.js

```diff
@@ -352,32 +353,28 @@ export function RegularElement(node, context) {
 
 	// special case — if an element that only contains text, we don't need
 	// to descend into it if the text is non-reactive
-	const is_text = trimmed.every((node) => node.type === 'Text' || node.type === 'ExpressionTag');
-
-	// in the rare case that we have static text that can't be inlined
-	// (e.g. `<span>{location}</span>`), set `textContent` programmatically
-	const use_text_content =
-		is_text &&
-		trimmed.every((node) => node.type === 'Text' || !node.metadata.expression.has_state) &&
-		trimmed.some((node) => node.type === 'ExpressionTag' && !node.metadata.expression.can_inline);
-
-	if (use_text_content) {
-		let { value } = build_template_chunk(trimmed, context.visit, child_state);
+	const states_and_calls =
+		trimmed.every((node) => node.type === 'Text' || node.type === 'ExpressionTag') &&
+		trimmed.some((node) => node.type === 'ExpressionTag') &&
+		get_states_and_calls(trimmed);
```

Comment:
I think we can preserve the essence of this change — it better articulates when and why we're using `textContent`, and allows us to get rid of the weirdo `get_states_and_calls` function

```suggestion
	// in the rare case that we have static text that can't be inlined
	// (e.g. `<span>{location}</span>`), set `textContent` programmatically
	const use_text_content =
		trimmed.every((node) => node.type === 'Text' || node.type === 'ExpressionTag') &&
		trimmed.every((node) => node.type === 'Text' || !node.metadata.expression.has_state) &&
		trimmed.some((node) => node.type === 'ExpressionTag');
```